src/core/Akka/Serialization/NewtonSoftJsonSerializer.cs

```diff
@@ -138,51 +172,75 @@ public NewtonSoftJsonSerializer(ExtendedActorSystem system, Config config)
             : this(system, NewtonSoftJsonSerializerSettings.Create(config))
         {
         }
-
-
+        
         public NewtonSoftJsonSerializer(ExtendedActorSystem system, NewtonSoftJsonSerializerSettings settings)
             : base(system)
         {
-            Settings = new JsonSerializerSettings
+            if (settings.UsePooledStringBuilder)
+            {
+                _sbPool = new DefaultObjectPool<StringBuilder>(
+                    new StringBuilderPooledObjectPolicy()
+                    {
+                        InitialCapacity = settings.StringBuilderMinSize,
+                        MaximumRetainedCapacity =
+                            settings.StringBuilderMaxSize
+                    });
+            }
+            Settings = CreateInternalSettings(system, settings,this);
+            var settingsNoFormat = CreateInternalSettings(system, settings,this);
+            settingsNoFormat.Formatting = Formatting.None;
+            _serializer = JsonSerializer.Create(Settings);
+            _formattingNoneSerializer = JsonSerializer.Create(settingsNoFormat);
+        }
+
+        private static JsonSerializerSettings CreateInternalSettings(
+            ExtendedActorSystem system, NewtonSoftJsonSerializerSettings settings, NewtonSoftJsonSerializer surrogateParent)
```

Comment:
I made this static because it makes it harder for people to accidentally miswire things (i.e. by accidentally applying to `Settings` on the instance when each created `JsonSerializerSettings` needs to have it's inits run.)